### ISSUE TITLE
Allow to modify item names (title)

### ISSUE THREAD

**Issue by bpminfotech** [frappe/books#518]
There is no Edit option for title ( Item Name) in sales or purchase of existing record.

**Comment by 18alantom** [comment]
This will be added later on, as of now Renaming is not available. You can _Duplicate_ the item and delete the old copy if there are no entries.

**Comment by bpminfotech** [comment]
Yeah, doing the same..Thanks..

**Comment by mildred** [comment]
I noticed that the name is used as a foreign key, which is not very good for renames. Perhaps instead a migration should be performed to change this to numeric ids, or the plan might be to update primary and foreign key in a single transaction?

**Comment by mildred** [comment]
The way renames was thought of was by updating all primary keys and foreign keys at once in the database in a single transaction.

This should be plugged here:

https://github.com/frappe/books/blob/4c789bbf26063dc76e7392d25953fe4e39f50793/backend/database/core.ts#L342-L351

It is called from here (if there is a need to pass in more parameters)

https://github.com/frappe/books/blob/4c789bbf26063dc76e7392d25953fe4e39f50793/fyo/model/doc.ts#L962-L973

There is already some code that goes through all of the links, ~~but perhaps it's not the right abstraction as it needs the database high-level JSON schema~~

https://github.com/frappe/books/blob/4c789bbf26063dc76e7392d25953fe4e39f50793/src/utils/doc.ts#L59-L62

(what's the TODO to normalize this function?)

**Comment by mildred** [comment]
we should allow to modify names for :

- Party
- Items
- Suppliers
- Purchase Items
- Customers
- Sale Items
- Accounts
- Tax
- Print Template

Probably the option should be made possible for all types of documents and the schema should state if the document should be allowed to be renamed.

**Comment by arunppsg** [comment]
It will also be great if item names in chart of account could be renamed as well as if regrouping is allowed.

**Comment by mildred** [comment]
Another option would be to have a display_name on the models and everywhere we show and search by the display_name. And renaming would be just changing the display_name...

**Comment by shyadav08** [comment]
I agree..This is a big limitation. Simply using the item names as primary identifier is not good. They should be using another (probably alphanumberic) identifier so as to allow a item name change. This is a basic functionality I expect as a user.
